File: validate.js
const express = require("express")
const app = express()
const bodyParser = require("body-parser")

app.use(bodyParser.json())
app.use(bodyParser.urlencoded({
    extended: true
}))

const credentials = {
    username: "YOUR_USERNAME_GOES_HERE",
    apiKey: "YOUR_API_KEY_GOES_HERE"
}

const AfricasTalking = require("africastalking")(credentials)

app.post("/payment-notification", async (req, res) => {
    const phoneNumber = req.body.source

    if (phoneNumber == "YOUR_PHONE_NUMBER_GOES_HERE") {
        response = {
            status: "Validated"
        }
    } else {
        response = {
            status: "Failed"
        }
    }

    return JSON.stringify(response)
})

app.listen(3000, () => {
    console.log("Server started...")
}

File: validate.py
from flask import Flask, request, jsonify

app = Flask(__name__)

phone_number = request.values.get("phoneNumber")

@app.route("/validate-transaction", methods=["GET", "POST"])
def validate_transaction():
    if (phone_number is "YOUR_PHONE_NUMBER_GOES_HERE"):
        response = {
            "status": "Validated"
        }
        return jsonify(response)
    else:
        response = {
            "status": "Failed"
        }
        return jsonify(response)
